sunpy/net/helioviewer.py

```diff
@@ -450,3 +433,18 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, source_id, key):
+        """Returns source_id based on the key"""
+        if source_id is None:
+            source_id_list = list(partial_match(key,self.data_sources))
+            if len(source_id_list) == 1:
+                source_id = source_id_list[0]
+                return source_id
+            else:
+                raise KeyError("The values used for observatory, instrument, detector, measurement "
+                               "do not correspond to a source_id. Please check the list using "
+                               "HelioviewerClient.data_sources. Expected 1 got " + str(len(source_id_list)) + 
+                               " values for source_id ", source_id_list)
```

Comment:
How does this error message look now?

Reply:
It is same as before. It looks like that only. I tried it

Reply:
If I recall, it needed rephrasing.

Reply:
I'm confused. What should it tell the user. I mean shall I make it more clear that the values passed by the user resulted in the following source_id list and then just show the list?
Or shall it just be a simple error saying that the key is wrong

Reply:
The user needs to be able to understand what went wrong and how to fix it. 
Printing out some source_id numbers does not help a user.

Reply:
So I think I have two issues with it as it stands. It is at one point really long and two does not give the user enough information to fix their issue.

Reply:
So maybe we want to change it something like:

```
                raise KeyError(f"The values used: {observatory}, {instrument}, {detector}, {measurement} "
                                "do not correspond to one source_id but str(len(source_id_list)) source_id(s)."
                                " Please check the list using HelioviewerClient.data_sources.")
```

Do you know if it's possible to get all the keys that partially match?

Reply:
Tho I don't think my version works since I have no accessed the right keys.

Reply:
Yeah it return all the values that matches partially

Reply:
But that isn't the keys but the values?

Reply:
Yes. It tries to get all the values that matches the partial key